R/utilities.R

```diff
@@ -805,181 +805,414 @@ trim_scale_expression = function(x){
-                                                   min_count_per_bin = 3,
-                                                   return_count = FALSE,
-                                                   drop_unmutated = FALSE,
-                                                   classification_column = "lymphgen",
-                                                   from_indexed_flatfile = FALSE,
-                                                   mode = "slms-3"){
-
-  max_region = 1000000
-  if(missing(metadata)){
-    metadata = get_gambl_metadata()
-  }
-  if(missing(this_region)){
-    this_region = paste0(chromosome, ":", start_pos, "-", end_pos)
-  }else{
-    chunks = region_to_chunks(this_region)
-    chromosome = chunks$chromosome
-    start_pos = as.numeric(chunks$start)
-    end_pos = as.numeric(chunks$end)
-  }
-  region_size = end_pos - start_pos
-  if(region_size < max_region){
-    message(paste("processing bins of size", window_size, "across", region_size, "bp region"))
-  }else{
+
+calc_mutation_frequency_sliding_windows <- function(this_region,
+                                          chromosome,
+                                          start_pos,
+                                          end_pos,
+                                          these_samples_metadata,
+                                          these_sample_ids,
+                                          maf = NULL,
+                                          projection = "grch37",
+                                          slide_by = 100,
+                                          window_size = 1000,
+                                          return_format = "long",
+                                          min_count_per_bin = 0,
+                                          return_count = TRUE,
+                                          drop_unmutated = FALSE,
+                                          from_indexed_flatfile = TRUE,
+                                          mode = "slms-3") {
+  # Create objects to describe region both as string and individual objects
+  try(if (missing(this_region) & missing(chromosome)) {
+    stop("No region information provided. Please provide a region as a string in the chrom:start-end format, or as individual arguments. ")
+  })
+
+  if ((drop_unmutated | min_count_per_bin > 0) & return_format == "wide") {
+    message("To return a wide table, all samples and windows must be kept. Ignoring drop_unmutated and min_count_per_bin arguments. ")
+  }
+
+  if (missing(this_region)) {
+    this_region <- paste0(
+      chromosome, ":", start_pos, "-",
+      end_pos
+    )
+  } else {
+    chunks <- GAMBLR:::region_to_chunks(this_region)
+    chromosome <- chunks$chromosome
+    start_pos <- as.numeric(chunks$start)
+    end_pos <- as.numeric(chunks$end)
+  }
+
+  # Check for provided metadata, else use GAMBL metadata
+  if (missing(these_samples_metadata)) {
+    metadata <- get_gambl_metadata(
```

Comment:
It might be risky to allow this function to work on samples of a different seq_type on one call to the function. If we make the user specify it at least they have to intentionally merge the output after, and hopefully also think about how it's likely not a good idea to do this in many scenarios.

Reply:
Another consideration here is those sample_id that have >1 seq type in GAMBL. Will this cause a problem in this function?

Reply:
So you think it's a good thing to allow users to run this function on a mix of exome data and genome data (possibly without realizing it)? That would be the default behaviour right now, wouldn't it?

Reply:
I addressed this by adding an argument to allow users to subset to one seq_type, but the default will remain the combination of capture and genome, utilizing the default seq_type_priority for get_gambl_metadata. 

Users should be encouraged to always provide metadata and not a vector of sample IDs so they can be confident that they're getting results only from the actual samples they want (I added a note about this to the id_ease documentation).